### DISCUSSION TITLE
Does WSL2 honor windows firewall settings?

### DISCUSSION THREAD

**Discussion by dlsharp** [microsoft/WSL#10094] [Q&A]
The following article came out a couple of years ago, and is the reason why our entire company is stuck on WSL1.

https://www.bleepingcomputer.com/news/microsoft/windows-subsystem-for-linux-2-bypasses-the-windows-10-firewall/

Has there been any progress on this front?  If this is no longer true, I would love to have something to point to that would allow us to argue that upgrading from WSL1 to WSL2 is appropriate.

**Comment by rjadidi920** [comment]
Hello?
Are the guys who developed WSL2 that dumb to forgotten about the fact that windows firewall rules should not be bypassed when they introduce a new feature to Windows?

**Reply by nlvw** [reply]
[https://devblogs.microsoft.com/commandline/windows-subsystem-for-linux-september-2023-update/](https://devblogs.microsoft.com/commandline/windows-subsystem-for-linux-september-2023-update/)

Feature has been added and is in preview/experimental.  This update came with a ton of really good networking fixes.
